cpp/src/parquet/platform.h

```diff
@@ -64,4 +74,38 @@
 #define PARQUET_TEMPLATE_EXPORT
 #endif
 
-#endif  // PARQUET_UTIL_VISIBILITY_H
+#define PARQUET_DISALLOW_COPY_AND_ASSIGN ARROW_DISALLOW_COPY_AND_ASSIGN
+
+#define PARQUET_NORETURN ARROW_NORETURN
+#define PARQUET_DEPRECATED ARROW_DEPRECATED
+
+// If ARROW_VALGRIND set when compiling unit tests, also define
+// PARQUET_VALGRIND
+#ifdef ARROW_VALGRIND
+#define PARQUET_VALGRIND
+#endif
+
+namespace parquet {
+
+namespace BitUtil = ::arrow::BitUtil;
+
+using Buffer = ::arrow::Buffer;
+using MemoryPool = ::arrow::MemoryPool;
+using MutableBuffer = ::arrow::MutableBuffer;
+using ResizableBuffer = ::arrow::ResizableBuffer;
+using ResizableBuffer = ::arrow::ResizableBuffer;
+using ArrowInputFile = ::arrow::io::RandomAccessFile;
+using ArrowInputStream = ::arrow::io::InputStream;
+using ArrowOutputStream = ::arrow::io::OutputStream;
+using string_view = ::arrow::util::string_view;
+
+constexpr int64_t kDefaultOutputStreamSize = 1024;
+
+std::shared_ptr<::arrow::io::BufferOutputStream> CreateOutputStream(
+    ::arrow::MemoryPool* pool = ::arrow::default_memory_pool());
```

Comment:
It seems that `PARQUET_EXPORT` is missing.

https://ci.appveyor.com/project/ApacheSoftwareFoundation/arrow/builds/24938354/job/8eqg024oo1mim9y8#L3103

> ```text
> bloom_filter-test.cc.obj : error LNK2019: unresolved external symbol "class std::shared_ptr<class arrow::io::BufferOutputStream> __cdecl parquet::CreateOutputStream(class arrow::MemoryPool *)" (?CreateOutputStream@parquet@@YA?AV?$shared_ptr@VBufferOutputStream@io@arrow@@@std@@PEAVMemoryPool@arrow@@@Z) referenced in function "private: virtual void __cdecl parquet::test::BasicTest_TestBloomFilter_Test::TestBody(void)" (?TestBody@BasicTest_TestBloomFilter_Test@test@parquet@@EEAAXXZ)
> ```

Reply:
Ah thanks. Fixing